esp/platform/application_config.cpp

```diff
@@ -0,0 +1,254 @@
+        if (xpath && xpath[0]=='@' && isupper(xpath[1]))
+            xpath++;
+        target->addProp(xpath, aiter->queryValue());
+    }
+    Owned<IPropertyTreeIterator> iter = toMerge->getElements("*");
+    ForEach (*iter)
+    {
+        IPropertyTree &e = iter->query();
+        target->addPropTree(e.queryName(), LINK(&e));
+    }
+}
+
+void addService(IPropertyTree *legacyEsp, IPropertyTree *app, const char *application, const char *service, const char *protocol, unsigned port, const char *auth, int seq)
+{
+    VStringBuffer plugin_xpath("service_plugins/@%s", service);
+    const char *service_plugin = app->queryProp(plugin_xpath);
+    VStringBuffer serviceXml("<EspService name='%s_service' type='%s' plugin='%s'/>", service, service, service_plugin);
+
+    VStringBuffer config_xpath("%s/%s", application, service);
+    IPropertyTree *serviceConfig = app->queryPropTree(config_xpath);
+    IPropertyTree *serviceEntry = legacyEsp->addPropTree("EspService", createPTreeFromXMLString(serviceXml));
+
+    if (serviceConfig && serviceEntry)
+        mergeServicePTree(serviceEntry, serviceConfig);
+
+    bindService(legacyEsp, app, service, protocol, port, auth, seq);
+}
+
+bool addProtocol(IPropertyTree *legacyEsp, IPropertyTree *app)
+{
+    bool useTls = app->getPropBool("@tls", true);
+    if (useTls)
+    {
+        StringBuffer protocolXml("<EspProtocol name='https' type='secure_http_protocol' plugin='esphttp' maxRequestEntityLength='60000000'/>");
+        IPropertyTree *protocol = legacyEsp->addPropTree("EspProtocol", createPTreeFromXMLString(protocolXml));
+        IPropertyTree *tls = app->queryPropTree("tls_config");
+        if (protocol && tls)
+        {
+            Owned<IAttributeIterator> aiter = tls->getAttributes();
+            ForEach (*aiter)
+                tls->addProp(aiter->queryName()+1, aiter->queryValue()); //attributes to elements to match legacy config
+            Owned<IPropertyTree> temp;
+            const char *name = app->queryProp("@name");
+            if (name)
+            {
+                StringBuffer xml;
+                toXML(tls, xml);
+                xml.replaceString("{$name}", name);
+                temp.setown(createPTreeFromXMLString(xml));
+                tls = temp.get();
+            }
+            mergePTree(protocol, tls);
+        }
+    }
+    else
+    {
+        StringBuffer protocolXml("<EspProtocol name='http' type='http_protocol' plugin='esphttp' maxRequestEntityLength='60000000'/>");
+        legacyEsp->addPropTree("EspProtocol", createPTreeFromXMLString(protocolXml));
+    }
+    return useTls;
+}
+
+void addServices(IPropertyTree *legacyEsp, IPropertyTree *appEsp, const char *application, const char *auth, bool tls)
+{
+    Owned<IPropertyTreeIterator> services = appEsp->getElements("application/services");
+    int defPort = tls ? appEsp->getPropInt("@tlsPort", 443) : appEsp->getPropInt("@httpPort", 80); //allows boolean switch between either http or tls config
```

Comment:
Not sure 80 is a good default. You can't bind to port 80 unless running as root or special capabilities set.

Reply:
Or perhaps httpPort=8010 should be included in the application=eclwatch and application=http_services yaml files,  so that the default port for those applications is what it has always been...